src/gcp/cloudfunctions.ts

```diff
@@ -260,6 +253,73 @@ export async function setIamPolicy(options: IamOptions): Promise<void> {
   }
 }
 
+/**
+ * Formats the service account to be used with IAM API calls, a vaild service account string is
+ * 'service-account@project.iam.gserviceaccount.com', 'service-account@', or 'service-account'.
+ * @param serviceAccount the custom service account created by the user
+ * @param projectId the ID of the current project
+ * @returns a correctly formatted service account string
+ *
+ * @throws {@link Error} if the supplied service account string is empty
+ */
+function formatServiceAccount(serviceAccount: string, projectId: string): string {
```

Comment:
`gcp/proto.ts` is becoming a grab bag of types useful for handling GCP interfaces. This could probably go there so you can reuse it in GCFv1 and Run

Reply:
Moved it over there